### DISCUSSION TITLE
Will react-router v7 support other bundlers than Vite ?

### DISCUSSION THREAD

**Discussion by Romej** [remix-run/react-router#12120] [General]
My understanding is that the next major version of react-router will mainly use a Vite plugin.

Otherwise, we may follow the [manual installation with vite](https://reactrouter.com/dev/guides/start/installation#manual-installation-with-vite).

But what if our codebase is older and we use webpack or rspack/rsbuild ? Will we be able to install v7 in some other way?

**Comment by sergiodxa** [comment]
You can use RR without Vite, you just won't get the things the a Vite plugin gives you. But the library itself is usable wit any bundler.

**Reply by mmeluch** [reply]
Can we get an actual answer to the question?

> Will react-router v7 support other bundlers than Vite _**without compromising on features**_?

Vite is still unusable for huge projects from a performance standpoint: https://github.com/vitejs/vite/discussions/13697#discussioncomment-7737220
